components/prism-core.js

```diff
@@ -235,6 +297,8 @@ var _ = self.Prism = {
 		var rest = grammar.rest;
 
 		if (rest) {
+			_.optimizeGrammar(rest);
```

Comment:
Yes, this should work. (But `concat` returns a new array so you must assign the result. And thus the property must be writable.)